Detectors/TOF/base/src/Geo.cxx

```diff
@@ -19,3 +20,478 @@ constexpr Float_t Geo::HEIGHTS[NPLATES][NMAXNSTRIP];
+    iPlate = 4;
+
+  return iPlate;
+}
+
+Int_t Geo::GetStrip(const Float_t* pos)
+{
+  //
+  // Returns the Strip index
+  //
+  constexpr Float_t HGLFY = HFILIY + 2 * HGLASSY; // heigth of GLASS+FISHLINE  Layer
+  constexpr Float_t HSTRIPY = 2. * HHONY + 2. * HPCBY + 4. * HRGLY + 2. * HGLFY + HCPCBY; // 3.11
+
+  Int_t iStrip = -1;
+
+  Float_t posLocal[3];
+  for (Int_t ii = 0; ii < 3; ii++)
+    posLocal[ii] = pos[ii];
+
+  Int_t isector = GetSector(posLocal);
+  if (isector == -1) {
+    return iStrip;
+  }
+  Int_t iplate = GetPlate(posLocal);
+  if (iplate == -1) {
+    return iStrip;
+  }
+
+  Int_t nstrips = 0;
+  switch (iplate) {
+    case 0:
+    case 4:
+      nstrips = NSTRIPC;
+      break;
+    case 1:
+    case 3:
+      nstrips = NSTRIPB;
+      break;
+    case 2:
+      nstrips = NSTRIPA;
+      break;
+  }
+
+  // ALICE reference frame -> B071/B074/B075 = BTO1/2/3 reference frame
+  Double_t angles[6] = { 90., 90. + (isector + 0.5) * PHISEC, 0., 0., 90., (isector + 0.5) * PHISEC };
+  Rotation(posLocal, angles);
+
+  Float_t step[3] = { 0., 0., static_cast<Float_t>((RMAX + RMIN) * 0.5) };
+  Translation(posLocal, step);
+
+  // B071/B074/B075 = BTO1/2/3 reference frame -> FTOA = FLTA reference frame
+  angles[0] = 90.;
+  angles[1] = 0.;
+  angles[2] = 0.;
+  angles[3] = 0.;
+  angles[4] = 90.;
+  angles[5] = 270.;
+
+  Rotation(posLocal, angles);
+
+  // FTOA/B/C = FLTA/B/C reference frame -> FSTR reference frame
+  Int_t totStrip = 0;
+  for (Int_t istrip = 0; istrip < nstrips; istrip++) {
+    Float_t posLoc2[3] = { posLocal[0], posLocal[1], posLocal[2] };
+
+    step[0] = 0.;
+    step[1] = GetHeights(iplate, istrip);
+    step[2] = -GetDistances(iplate, istrip);
+    Translation(posLoc2, step);
+
+    if (GetAngles(iplate, istrip) > 0.) {
+      angles[0] = 90.;
+      angles[1] = 0.;
+      angles[2] = 90. + GetAngles(iplate, istrip);
+      angles[3] = 90.;
+      angles[4] = GetAngles(iplate, istrip);
+      angles[5] = 90.;
+    } else if (GetAngles(iplate, istrip) == 0.) {
+      angles[0] = 90.;
+      angles[1] = 0.;
+      angles[2] = 90.;
+      angles[3] = 90.;
+      angles[4] = 0;
+      angles[5] = 0.;
+    } else if (GetAngles(iplate, istrip) < 0.) {
+      angles[0] = 90.;
+      angles[1] = 0.;
+      angles[2] = 90. + GetAngles(iplate, istrip);
+      angles[3] = 90.;
+      angles[4] = -GetAngles(iplate, istrip);
+      angles[5] = 270.;
+    }
+    Rotation(posLoc2, angles);
+
+    if ((TMath::Abs(posLoc2[0]) <= STRIPLENGTH * 0.5) && (TMath::Abs(posLoc2[1]) <= HSTRIPY * 0.5) &&
+        (TMath::Abs(posLoc2[2]) <= WCPCBZ * 0.5)) {
+      iStrip = istrip;
+      totStrip++;
+      for (Int_t jj = 0; jj < 3; jj++)
+        posLocal[jj] = posLoc2[jj];
+      break;
+    }
+  }
+
+  return iStrip;
+}
+
+Int_t Geo::GetPadZ(const Float_t* pos)
+{
+  //
+  // Returns the Pad index along Z
+  //
+
+  Int_t iPadZ = -1;
+
+  Float_t posLocal[3];
+  for (Int_t ii = 0; ii < 3; ii++)
+    posLocal[ii] = pos[ii];
+
```

Comment:
Hi Ruben,
I put all the single methods (GetSector, GetPlate, ...) as private leaving only the GetID as a public method. In this way I can force the different indexes to be comupted in the right order calling rotations/translation only once.